pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -724,12 +762,55 @@ void registerBackendAgnosticAlgorithms(nb::module_ &m) {
  */
 template <class StateVectorT, class PyClass>
 void registerBackendAgnosticStateVectorMethods(PyClass &pyclass) {
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    using ComplexT = typename StateVectorT::ComplexT;
     // Initialize with number of qubits
     pyclass.def(nb::init<size_t>());
 
     pyclass.def("__len__", &StateVectorT::getLength,
                 "Get the size of the statevector.");
     pyclass.def("size", &StateVectorT::getLength);
+
+    // Reset state vector - common across all backends
+    pyclass.def("resetStateVector", &StateVectorT::resetStateVector,
+                "Reset the state vector to |0...0>.");
+
+    // Set basis state - with conditional for async parameter (LGPU)
+    pyclass.def(
+        "setBasisState",
+        [](StateVectorT &sv, const std::vector<std::size_t> &state,
+           const std::vector<std::size_t> &wires, const bool async) {
+            if constexpr (requires { sv.setBasisState(state, wires, async); }) {
+                sv.setBasisState(state, wires, async);
+            } else {
+                sv.setBasisState(state, wires);
+            }
+        },
+        "Set the state vector to a computational basis state.",
+        nb::arg("state") = nullptr, nb::arg("wires") = nullptr,
```

Comment:
Shouldn't we use `nb::none()` instead of standard c++ `nullptr?

Reply:
probably yes